internal/command/command.go

```diff
@@ -694,3 +697,17 @@ func ChangeWorkingDirectory(ctx context.Context, wd string) (context.Context, er
 
 	return state.WithWorkingDirectory(ctx, wd), nil
 }
+
+func setUsingGPU(ctx context.Context) (context.Context, error) {
+	appConfig := appconfig.ConfigFromContext(ctx)
+	if appConfig == nil {
+		instrument.UsingGPU = false
+		return ctx, nil
+	}
+
+	instrument.UsingGPU = lo.SomeBy(appConfig.Compute, func(x *appconfig.Compute) bool {
+		return x != nil && x.MachineGuest != nil && x.MachineGuest.GPUKind != ""
+	})
```

Comment:
ignore me but had to suggest a simplification nitpick:

```suggestion
	if appConfig != nil {
		instrument.UsingGPU = lo.SomeBy(appConfig.Compute, func(x *appconfig.Compute) bool {
			return x != nil && x.MachineGuest != nil && x.MachineGuest.GPUKind != ""
		})
	}
```